Recruit-Training/Basic-Recruit-Training/person/Person.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Person.h"
+
+Person::Person(const std::string &name, int age) : name_(name), age_(new int(age))
+{};
+
+Person::Person()
+{};
+
+Person::~Person()
+{
+    delete age_;
+}
+
+int Person::getAge() const
+{
+    return *age_;
+}
+
+const std::string& Person::getName() const
+{
+    return name_;
+}
+
+void Person::setName(const std::string& newName)
+{
+    name_ = newName;
+}
+
+void Person::setAge(int newAge)
+{
+    delete age_;
+    age_ = new int(newAge);
```

Comment:
This probably works but you can just dereference age_ and assign it to newAge. When you allocate memory for a variable you can change the value stored in that memory address. If you call setAge a bunch of time, constantly deleting and re allocating is likely going to be pretty expensive.